service/history/retry.go

```diff
@@ -93,18 +93,24 @@ func isRetryable(failure *failurepb.Failure, nonRetryableTypes []string) bool {
 		return false
 	}
 
-	if failure.GetApplicationFailureInfo().GetNonRetryable() || failure.GetServerFailureInfo().GetNonRetryable() {
-		return false
-	}
-
 	if failure.GetTimeoutFailureInfo() != nil {
 		if failure.GetTimeoutFailureInfo().GetTimeoutType() != enumspb.TIMEOUT_TYPE_START_TO_CLOSE &&
```

Comment:
[Nit] can we just do return TimeoutType() == START_TO_CLOSE || TimeoutType() == TIMEOUT_TYPE_HEARTBEAT

Reply:
I am so used to golang not having `?:` operator, so I forgot that such returns are actually possible.